### PR TITLE
[MSHADE-425] Relocate services name before add to serviceEntries

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSHADE) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHADE-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHADE-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @khmarbaise @rmannibucau

**Comment by slawekjaranowski** [comment]
@tisonkun Thanks for your PR.
We need a unit test to be sure that next change don't break it.

**Comment by tisonkun** [comment]
@slawekjaranowski Yes. I ever thought of this. Let me try to dig out how to add one. Or could you show me some related tests.

**Comment by slawekjaranowski** [comment]
I don't investigate deeper - only mention about missing test

**Comment by slawekjaranowski** [comment]
There is a https://github.com/apache/maven-shade-plugin/blob/master/src/test/java/org/apache/maven/plugins/shade/resource/ServiceResourceTransformerTest.java

maybe tests don't cover all path - you change something in class but tests still works without change

**Comment by tisonkun** [comment]
@slawekjaranowski Yes. I would say this patch is simple enough to be correct. But I'm glad to add some tests to prevent regressions.

**Comment by tisonkun** [comment]
@slawekjaranowski Added at d29c633cea770ef59464e0b523a938bcd7590c50. Also verify locally that without this patch the test failed.

**Review by rmannibucau — APPROVED** [review]

File: src/test/java/org/apache/maven/plugins/shade/resource/ServiceResourceTransformerTest.java [hunk 1]
```diff
@@ -85,6 +85,49 @@ public void relocatedClasses() throws Exception {
...
         }
     }
+
+    @Test
+    public void mergeRelocatedFiles() throws Exception {
+        SimpleRelocator relocator =
+                new SimpleRelocator( "org.foo", "borg.foo", null, Arrays.asList( "org.foo.exclude.*" ) );
+        relocators.add( relocator );
+
+        String content = "org.foo.Service\norg.foo.exclude.OtherService\n";
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/plugins/shade/resource/ServiceResourceTransformerTest.java:95]: Why not use constant `NEWLINE` as is used later

Comment (tisonkun) [reply src/test/java/org/apache/maven/plugins/shade/resource/ServiceResourceTransformerTest.java:95]: Make sense. I just copy from the test above. Code refactoring can be another topic that I don't mess up one PR. But it's fair enough for new code :)

Comment (tisonkun) [reply src/test/java/org/apache/maven/plugins/shade/resource/ServiceResourceTransformerTest.java]: Resolved.

**Comment by rmannibucau** [comment]
@tisonkun on my side I was just waiting for @slawekjaranowski to ack (or merge, no issue) but otherwise no blocker for this pr for me

**Comment by slawekjaranowski** [comment]
you can squash to one final commit and will be ok for me also.

**Comment by tisonkun** [comment]
Squashed.

Though, you can use GitHub "Squash and Commit" button to perform the squash, so that we don't need one more CI turn.

**Comment by tisonkun** [comment]
@rmannibucau @slawekjaranowski I may a bit in a hurry but I'd like to know whether/when we can have a release for this fix. It resolves one or several downstream use case and I'm willing to upgrade for this fix.

**Comment by rmannibucau** [comment]
@tisonkun guess a mail on the mailing list can make people more aware of it (at least the ones with the full karma to do it ;))

**Comment by rmannibucau** [comment]
@slawekjaranowski nop, only a PMC can achieve it (for goods) so if a commiter can start it he can't end it so always better to ensure some PMC is there to handle all the needed steps before starting by XP.

**Comment by slawekjaranowski** [comment]
if you wish we can discuss on dev list
look last pending release of m-pmd-p

**Comment by tisonkun** [comment]
Thanks! I'm subscribing dev@maven.apache.org mailing list now.

> m-pmd-p

Out of curiosity, what does this stand for?

**Comment by slawekjaranowski** [comment]
m-pmd-p -> maven-pmd-plugin 😄

**Comment by jira-importer** [comment]
Resolve #702

**Pull request merged**